src/Tribe/Importer/File_Importer_Events.php

```diff
@@ -10,27 +10,55 @@ class Tribe__Events__Importer__File_Importer_Events extends Tribe__Events__Impor
 	protected function match_existing_post( array $record ) {
 		$start_date = $this->get_event_start_date( $record );
 		$end_date   = $this->get_event_end_date( $record );
+		$all_day    = $this->get_boolean_value_by_key( $record, 'event_all_day' );
+
+		// Base query - only the meta query will be different
 		$query_args = array(
 			'post_type'      => Tribe__Events__Main::POSTTYPE,
 			'post_title'     => $this->get_value_by_key( $record, 'event_name' ),
-			'meta_query'     => array(
+			'fields'         => 'ids',
+			'posts_per_page' => 1,
+		);
+
+		// When trying to find matches for all day events, the comparison should only be against the date
+		// component only since a) the time is irrelevant and b) the time may have been adjusted to match
+		// the eod cutoff setting
+		if ( in_array( $all_day, array( 'yes', 'true', '1' ) ) ) {
```

Comment:
Well, we could do that - I just think it's taking us in the wrong direction. 

I feel what we really need is a separate `is_true()` type helper function that tests for truthiness and which can be localized (if we have one, I've missed it) which is important in this scenario, where information is coming from a CSV source we don't control and where users might reasonably want to use keywords in a language of their choice.

This would be more appropriate than `::is_all_day()` which is very specific, tests for values we control and cannot be localized (and since it is testing against values we set, why should it be?).

Right now I need to match the array defined in the signature of `Tribe__Events__Importer__File_Importer_Events::get_boolean_value_by_key()` which I can't access by other means. I do feel that needs to be addressed, but that it's a separate problem.